xarray/backends/api.py

```diff
@@ -298,13 +307,19 @@ def _chunk_ds(
     chunks,
     overwrite_encoded_chunks,
     inline_array,
+    from_array_kwargs,
     **extra_tokens,
 ):
-    from dask.base import tokenize
+    if from_array_kwargs["manager"] == "dask":
+        from dask.base import tokenize
 
-    mtime = _get_mtime(filename_or_obj)
-    token = tokenize(filename_or_obj, mtime, engine, chunks, **extra_tokens)
-    name_prefix = f"open_dataset-{token}"
+        mtime = _get_mtime(filename_or_obj)
+        token = tokenize(filename_or_obj, mtime, engine, chunks, **extra_tokens)
+        name_prefix = f"open_dataset-{token}"
+    else:
+        # not used
+        token = (None,)
+        name_prefix = None
```

Comment:
This dask-specific code block is irritating because I would like to move it inside the `DaskManager._from_array()` method (and pass all the dask-specific kwargs inside `from_array_kwargs`), but the tokenization is done differently depending on where in the codebase `.chunk` is called from. I'm struggling to think of a simple pattern that would allow me to get around this path-dependence.

If I'm allowed to change the name of the dask objects created then I can refactor this more easily, but that breaks `TestDataArrayAndDataset.test_new_chunk`.